scripts/index.js

```diff
@@ -1 +1,153 @@
-console.log(`There are ${characters.length} characters in the array.`);
\ No newline at end of file
+console.log(`There are ${characters.length} characters in the array.`);
+
+//For each of these questions, write a function that returns the answer.
+
+//How many characters names start with "A"?
+
+function char_a(characters) {
+    let char_a = [];
+    characters.forEach(function (character) {
+        if (character.name[0] === "A" && character != "") {
+        char_a.push(character.name)
+        };
+    })
+    console.log(char_a)
+}
+
+// Output 168 + list of names
+//===============================================//
+
+//How many characters names start with "Z"?
+function char_z(characters) {
+    let char_z = [];
+    characters.forEach(function (character) {
+        if (character.name[0] === "Z" && character != "") {
+        char_z.push(character.name)
+        };
+    })
+    console.log(char_z)
+}
+// output 8 + list of names
+//===============================================//
+
+
+//How many characters are dead?
+function char_dead(characters) {
+    let char_dead = [];
+    characters.forEach(function (character) {
+        if (character.died !== "") {
+        char_dead.push(character.name)
+        };
+    })
+    console.log(char_dead)
+}
+
+// output 553 + list of names
+//===========================================//
+
+
+//Who has the most titles?
+function title_length(characters) {
+    let max_titles = [];
+        for (character in characters) {
+        if ((character.title).length >= 0) {
+            max_titles.push(character.title)
+        };
+    console.log(max_titles)
+    }
+}
+///// UGH!!!! ////
+//========================================//
+
+//How many are Valyrian?
+function valyrian(characters) {
+    let valyrian_names = [];
+    characters.forEach(function(character) {
+        if (character.culture === "Valyrian") {
+            valyrian_names.push(character.name)
+        }
+    });
+    console.log (valyrian_names)
+    }
+// Output 57 + list of names
+//========================================//
+
+//What actor plays "Hot Pie" (and don't use IMDB)?
+
+function whoIsHotPie(characters) {
+    const arrayOfHotPies = characters.filter(function (person) {
+        return person.name === "Hot Pie"
+    });
+    //console.log(arrayOfHotPies)
+    if (arrayOfHotPies.length > 0) {
+        const hotPie = arrayOfHotPies[0];
+        return (hotPie.playedBy);
+    } else {
+        return ('There are no Hot Pies');
+    }
+}
+whoIsHotPie(characters)
+// returns undefined even after following in class //
+function findActorFor2(characters, characterName) {
+    const theCharacter = characters.find(function(person) {
+        return person.name === characterName;
+    });
+    return theCharacter
+}
+
+findActorFor2(characters)
```

Comment:
Nice job on this. The only thin missing is the second argument to your function. It expects two arguments but we've only passed it on so characterName at this point would be undefined. Just add a name you'd like to a find and as long as it in the data structure you should be good to go. Great work again.